GHermeneus/include/GHermeneus/Machine.h

```diff
@@ -5,4 +5,185 @@
+                {
+                    os << " " << param.param << " = " << std::to_string(param.value);
+                }
+                os << std::endl;
+            }
+            return os;
+        };
+
+        /*!
+         * @brief Output the GCode stream to the Machines Instruction Vector
+         * @param machine The machine of type Machine<SSV_T, T>
+         * @param GCode the GCode as a string_view stream
+         * @return A machine of type Machine<SSV_T, T>
+         */
+        friend Machine<SSV_T, T>& operator<<(Machine<SSV_T, T>& machine, const std::string_view& GCode)
+        {
+            machine.parse(GCode);
+            return machine;
+        };
+
+    private:
+
+        /*!
+         * @brief extract the individual lines from the GCode
+         * @param GCode a string_view containing the GCode
+         */
+        void extractLines(const std::string_view& GCode)
+        {
+            // TODO: keep in mind CRLF and LF
+            lines = gcode
+                    | ranges::views::split('\n')
+                    | ranges::views::transform([](auto&& line) {
+                        return std::string_view(&*line.begin(), ranges::distance(line));})
+                    | ranges::views::enumerate
+                    | ranges::to_vector;
+        };
+
+        /*!
+         * @brief Extract an instruction of type Instruction<SSV_T, T> from a GCode line. If the line contains no
+         *  instruction it is a std::nullopt
+         * @param GCodeline string_view containing the line with command, the relevant parameters and/or a comment.
+         * @return an optional instruction of type Instruction<SSV_T, T>
+         */
+        [[nodiscard]] static std::optional<Instruction<SSV_T, T>> extractCmd(const Line& GCodeline)
+        {
+#ifndef NDEBUG
+            std::cout << GCodeline.first << ": " << GCodeline.second << std::endl; // Todo: Use a proper logger
+#endif
+            auto &&[lineno, gcode] = GCodeline;
+
+            // Split line in instruction and comment
+            auto split_line = gcode | ranges::views::split(';');
+
+            // Split the instruction into individual words
+            auto split_instruction = *split_line.begin()
+                                     | ranges::views::split(' ');
+
+            // Get the Cmd
+            auto cmd_view = split_instruction
+                       | ranges::views::take(1)
+                       | ranges::views::transform([&](auto&& c) {
+                            return std::string_view(&*c.begin(), ranges::distance(c));});
+            std::string_view cmd = *cmd_view.begin();
+            if (cmd.empty()) // No cmd nothing to do here Todo: also take into account comments that should be treated as a cmd
+            {
+                return std::nullopt;
+            }
+
+            // Get the values and parameters
+            // TODO: use std::from_char instead of copying to string
+            auto params = split_instruction
```

Comment:
What happens to a nonsense (or 'wrong dialect') line?

Reply:
This is parsed at this time in the program, but not mapped to an actual state space vector because the nonsense string_view has no key in 'Translator` class.
I still should account for nonsense that have a key by accident but wrong parameters though. But that will probably also be handled by the `Translator` class.
I will add a TODO to the code in that class as a reminder 😉